src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesEdit.js

```diff
@@ -94,10 +97,16 @@ export default class ExternalDataSourcesEdit extends React.Component {
               dataOptions={[
                 { value: 'org.olf.kb.adapters.GOKbOAIAdapter', label: 'org.olf.kb.adapters.GOKbOAIAdapter' }
               ]}
+              disabled={isReadonly}
               label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.type" />}
               name={`${name}.type`}
-              required
-              validate={required}
+              required={!isReadonly}
+              validate={v => {
+                if (!isReadonly) {
+                  return required(v);
```

Comment:
Maybe use this opportunity to remove our dependence on this module's local version of `required` and just use the stripes-erm-components copy.

Reply:
✔️